### ISSUE TITLE
run smoke test on snap

### ISSUE THREAD

**Issue by gares** [coq/platform#208]
In order to do so `shell_scripts/create_smoke_test_kit.sh` should be patched to be able to take package names and versions from a pick, rather than from the current opam state (installed package).

This is because the snap is built in a container, and the only output is a snap.
I need to run a task (outside the container) that builds the smoke test from the same pick without installing all packages.
Then install the snap and run the smoke test.

@MSoegtropIMC does this make sense to you? Could you implement this?

**Comment by MSoegtropIMC** [comment]
You need opam anyway to get the sources. I think it makes most sense to create the smoke test kit when the snap file is created - opam is available then (it is not really needed any more, but I left it in expecting something like this).

**Comment by gares** [comment]
opam yes, but the scripts looks for *installed* `coq-*` packages. Installing all of them requires running the platform scripts (again).

**Comment by MSoegtropIMC** [comment]
> but the scripts looks for installed coq-* packages

ah, ok, thats what you mean. Makes sense.

**Comment by MSoegtropIMC** [comment]
Btw.: one could also merge the snap run with the ubuntu run and use the smoke test kit produced by ubuntu.

**Comment by gares** [comment]
Yes, this is what i tried, but the pick is different... so the smoke would be off

**Comment by MSoegtropIMC** [comment]
Ubuntu builds all picks and all smoke tests (it just doesn't upload them as it looks ...)

**Comment by gares** [comment]
I missed that. Then we could upload them with a name which contains the pick and download one in a third job which pulls the snap and the appropriate smoke kit. I'll try that on Monday I guess.

**Comment by MSoegtropIMC** [comment]
It would likely make the upload task more tricky. I recently had a (yet untested) idea to make matrix entries depend on - possibly complex - conditions. The idea to run the dev build only on scheduled jobs would look like this:
```
    strategy:
      fail-fast: false
      matrix:
        variant:
          - '8.14~2022.01'
          - '8.13~2022.01'
          - '8.12'
          - '8.15~beta1'
          - 'dev'
        shouldRunDev:
          - ${{ github.event_name == 'schedule' }}
        exclude:
          - shouldRunDev: false
            variant: dev
```
As I said this is yet untested but should work according to docs. The trick is to have a matrix variable which is defined as true or false depending on a condition and then one excludes the case that the condition is false.

**Comment by MSoegtropIMC** [comment]
This is fixed by commit 23264b632eaec8b183f168f8f1f16e34319a40c6.
